### PR TITLE
[fix] move `parquet_stats_skipping` to `predicates` module

### PR DESCRIPTION
**Pull request by zachschuermann**
Our builds are actually broken due to a dependency on `mod engine` in `scan/mod.rs` (which only exists when one of the engine feature-flags is enabled. as soon as all are off, it fails to build). This fixes the issue by moving the `parquet_stats_skipping` module out of engine code and into `expressions` module. This was trivial since it actually doesn't depend on any engine code.

We pull `parquet_stats_skipping` into `scan/mod.rs` in order to `impl ParquetStatsProvider for NoStats`. Now instead of having this rely on engine code, it's moved to the `predicates` module (which is always included)

Note that I have a new test in CI that will catch this failure in the future: #601

### CODE REVIEW

**Comment by zachschuermann** [comment]
not sure why there were so many formatting changes..?

File: kernel/src/expressions/parquet_stats_skipping/tests.rs
Comment (zachschuermann) [inline kernel/src/expressions/parquet_stats_skipping/tests.rs:1]: all changes are just formatting - not sure why they are just now getting applied?

File: kernel/src/expressions/parquet_stats_skipping.rs
Comment (zachschuermann) [inline kernel/src/expressions/parquet_stats_skipping.rs:1]: all changes are just formatting - not sure why they are just now getting applied?

**Review by nicklan — APPROVED** [review]
lgtm!

**Review by nicklan — APPROVED** [review]

**Review by hntd187 — APPROVED** [review]
LGMT

File: kernel/src/predicates/parquet_stats_skipping/tests.rs [hunk 1]
```diff
@@ -236,7 +234,7 @@ impl ParquetStatsProvider for NullCountTestFilter {
 fn test_eval_is_null() {
     let expressions = [
         Expr::is_null(column_expr!("x")),
-        !Expr::is_null(column_expr!("x"))
+        !Expr::is_null(column_expr!("x")),
```
Comment (OussamaSaoudi-db) [inline kernel/src/predicates/parquet_stats_skipping/tests.rs:237]: huh Im surprised formatting ended up adding this comma

**Review by OussamaSaoudi-db — APPROVED** [review]

**Pull request merged**